### PR TITLE
Add extra version of py-matplotlib; fix dependency on six

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-matplotlib/package.py [hunk 1]
```diff
@@ -82,14 +83,16 @@ class PyMatplotlib(PythonPackage):
     depends_on('py-setuptools', type=('build', 'run'))  # See #3813
     depends_on('py-cycler@0.10:', type=('build', 'run'))
     depends_on('py-python-dateutil@2.1:', type=('build', 'run'))
+    depends_on('py-kiwisolver@1.0.1:', type=('build', 'run'), when='@2.2.0:2.2.4')
     depends_on('py-kiwisolver@1:', type=('build', 'run'), when='@2.2.0:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-matplotlib/package.py]: This kiwisolver business is confusing. `INSTALL.rst` has always said `>= 1.0.0`, but `setupext.py` and `lib/matplotlib.egg-info/requires.txt` have always said `>= 1.0.1`. To be on the safe side, I would remove the line you added above and use:
```suggestion
    depends_on('py-kiwisolver@1.0.1:', type=('build', 'run'), when='@2.2.0:')
```

File: var/spack/repos/builtin/packages/py-matplotlib/package.py [hunk 2]
```diff
@@ -82,14 +83,16 @@ class PyMatplotlib(PythonPackage):
...
     depends_on('py-pyparsing', type=('build', 'run'))
     depends_on('py-pytz', type=('build', 'run'), when='@:2')
     depends_on('py-subprocess32', type=('build', 'run'), when='^python@:2.7')
     depends_on('py-functools32', type=('build', 'run'), when='@:2.0.999 ^python@2.7')
     depends_on('py-backports-functools-lru-cache', type=('build', 'run'),
                when='@2.1.0:2.999.999 ^python@:2')
-    depends_on('py-six@1.9.0:', type=('build', 'run'), when='@:2')
+    depends_on('py-six@1.10.0:', type=('build', 'run'), when='@2:')
+    depends_on('py-six@1.9.0:',  type=('build', 'run'), when='@:2')
```
[the remark this answers is not in the document]
Comment (iarspider) [reply var/spack/repos/builtin/packages/py-matplotlib/package.py]: The version ranges in Spack confuse me a lot: sometimes it looks like the upper-bound is included, sometimes - it is not. Sometimes I see that "2" is used as a synonym for "2.0", "2.0.0" etc - and now I learn that it is a mask :(

File: var/spack/repos/builtin/packages/py-matplotlib/package.py [hunk 3]
```diff
@@ -82,14 +83,15 @@ class PyMatplotlib(PythonPackage):
...
-    depends_on('py-kiwisolver@1:', type=('build', 'run'), when='@2.2.0:')
+    depends_on('py-kiwisolver@1.0.1:', type=('build', 'run'), when='@2.2.0:')
     depends_on('py-pyparsing', type=('build', 'run'))
     depends_on('py-pytz', type=('build', 'run'), when='@:2')
     depends_on('py-subprocess32', type=('build', 'run'), when='^python@:2.7')
     depends_on('py-functools32', type=('build', 'run'), when='@:2.0.999 ^python@2.7')
     depends_on('py-backports-functools-lru-cache', type=('build', 'run'),
                when='@2.1.0:2.999.999 ^python@:2')
-    depends_on('py-six@1.9.0:', type=('build', 'run'), when='@:2')
+    depends_on('py-six@1.10.0:', type=('build', 'run'), when='@2:2.999.999')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-matplotlib/package.py]:
```suggestion
    depends_on('py-six@1.10.0:', type=('build', 'run'), when='@2.0:2.999')
```

File: var/spack/repos/builtin/packages/py-matplotlib/package.py [hunk 4]
```diff
@@ -82,14 +83,15 @@ class PyMatplotlib(PythonPackage):
...
+    depends_on('py-six@1.9.0:',  type=('build', 'run'), when='@:1.999.999')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-matplotlib/package.py]:
```suggestion
    depends_on('py-six@1.9.0:',  type=('build', 'run'), when='@:1')
```

**Pull request merged**